x-pack/plugins/security_solution/public/common/components/event_details/get_alert_summary_rows.tsx

```diff
@@ -395,3 +404,35 @@ function enrichThresholdCardinality(
     };
   }
 }
+
+/**
+ * if new terms fields present and have the same length as new terms
+ * transform new terms in following format:
+ * new_terms_fields: ['field-1', 'field-2']
+ * new_terms: ['value-1', 'value-2']
+ * enriched new_terms: ['field-1: value-1', 'field-2: value-2']
+ */
+function enrichNewTerms(
+  { values: newTerms }: TimelineEventsDetailsItem,
+  data: TimelineEventsDetailsItem[],
+  description: EnrichedFieldInfo
+) {
+  const newTermsFields = data.find(({ field }) => field === ALERT_NEW_TERMS_FIELDS)?.values;
+
+  // terms values and terms fields arrays must have the same length
+  if (
+    !Array.isArray(newTerms) ||
+    !Array.isArray(newTermsFields) ||
+    newTerms.length !== newTermsFields.length
```

Comment:
As a random question, is there ever a possibility where they aren't the same length?

Reply:
Not one I can think of. But I wanted to make sure I cover it as well, as it's essential to this use case the length of both arrays is the same